### PR TITLE
Updated Makefile

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

**What this PR does / why we need it**:

- Added linting and depcheck for frontend
- Throws error for all linting and dependency failures

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary?
* [ ] Have you added task comments where necessary?
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any?
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure?
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks?
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted?

### CODE REVIEW

**Comment by rajdas98** [comment]
Depcheck should not be in frontend code. It should in circleci

**Comment by gdsoumya** [comment]
> Depcheck should not be in frontend code. It should in circleci

It is in circleci, it is added to npm scripts, just like npm run lint.

File: litmus-portal/Makefile [hunk 1]
```diff
@@ -68,41 +68,60 @@ litmus-portal-push:
...
 .PHONY: litmus-portal-test
-litmus-portal-test : frontend-unit-test
+litmus-portal-test : npm-install frontend-unit-test
 
 frontend-unit-test:
 	@echo "------------------"
 	@echo "--> Frontend Unit Test"
 	@echo "------------------"
-	cd ./frontend && npm i && npx cypress run
+	cd ./frontend && npx cypress run
```
Comment (arkajyotiMukherjee) [inline litmus-portal/Makefile:127]: after PR #1831 we can just run `cypress run`
cc @Jonsy13

File: litmus-portal/Makefile [hunk 2]
```diff
@@ -68,41 +68,60 @@ litmus-portal-push:
...
+		echo "gofmt checking failed!" && echo "$${fmtRes}" \
+		&& echo "Please ensure you are using $$($(GO) version) for formatting code." \
+		&& exit 1;\
 	fi
 
 unused-package-check:
+	@echo "------------------"
+	@echo "--> Check litmus-portal frontend [depcheck]"
+	@echo "------------------"
+	cd frontend && npm i depcheck --no-save && node_modules/.bin/depcheck .
```
Comment (arkajyotiMukherjee) [inline litmus-portal/Makefile:99]: Does this skip the missing dependency checks as it was not giving correct/desired output

Comment (arkajyotiMukherjee) [inline litmus-portal/Makefile:99]: does it skip missing dependency checks as it was showing incorrect/undesirable output

Comment (gdsoumya) [reply litmus-portal/Makefile:99]: No, I'll make that change

**Pull request merged**